File: backend/src/main/java/com/gxs/bank/repository/BillPaymentRepository.java
package com.gxs.bank.repository;

import com.gxs.bank.model.BillPayment;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.List;
import java.util.UUID;

@Repository
public interface BillPaymentRepository extends JpaRepository<BillPayment, UUID> {
    List<BillPayment> findByUserId(UUID userId);
}

File: backend-python/com/gxs/bank/repository/BillPaymentRepository.py
from com.gxs.bank.model.BillPayment import BillPayment
from com.gxs.bank.repository._base import BaseRepository


class BillPaymentRepository(BaseRepository):
    def __init__(self, db):
        super().__init__(db, BillPayment)

    def findByUserId(self, userId: str):
        return self.db.query(BillPayment).filter(BillPayment.userId == userId).all()
